### PR TITLE
Feature/boilerplate

### PR DESCRIPTION
**Pull request by fran-costa**
Pull request for isssue #1 Create boilerplate

### CODE REVIEW

File: beautifulthings/README [hunk 1]
```diff
@@ -0,0 +1,8 @@
+To integrate the react app into Cordova app, follow these instructions:
+
+1.  Delete all files from <cordova-root-dir>/www directory.
+2.  Edit the file <react-root-dir>/src/index.js to wrap the main logic
+    inside a function to call it after deviceready CordovaEvent.
+3.  Edit <react-root-dir>/public/index.html file to include cordova.js
+    file.
+
```
Comment (chiiph) [inline beautifulthings/README]: You should probably remove this README

**Comment by ivanalejandro0** [comment]
I don't think it makes much sense to have a `beautifulthings/` folder inside the repo, which is named `beautifulthings` as well. We will end up with `beautifulthings/beautifulthings/` most of the time.

File: beautifulthings/react-app/src/index.js [hunk 2]
```diff
@@ -4,5 +4,13 @@ import './index.css';
...
 import registerServiceWorker from './registerServiceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
-registerServiceWorker();
+const startApp = () => {
+    ReactDOM.render(<App />, document.getElementById('root'));
+    registerServiceWorker();
+}
+
+if( !window.cordova ) {
```
[the remark this answers is not in the document]
Comment (ivanalejandro0) [reply beautifulthings/react-app/src/index.js]: +1
We should add `eslint` into the mix eventually.

**Comment by ivanalejandro0** [comment]
Does this work for you?

I needed to set the `homepage` for the CRA in order for this to launch on mobile.
Also, once you run the app, the platform and a plugin are added to the `package.json`, we should commit that.

Here's what I've changed to make it work:

```diff
diff --git a/config.xml b/config.xml
index fc0e65b..35c16dc 100644
--- a/config.xml
+++ b/config.xml
@@ -23,4 +23,5 @@
         <allow-intent href="itms:*" />
         <allow-intent href="itms-apps:*" />
     </platform>
+    <engine name="android" spec="^7.0.0" />
 </widget>
diff --git a/package.json b/package.json
index 07f9b94..ee8b5f8 100644
--- a/package.json
+++ b/package.json
@@ -10,5 +10,17 @@
         "run": "node scripts/react-to-cordova.js run"
     },
     "author": "Apache Cordova Team",
-    "license": "Apache-2.0"
-}
+    "license": "Apache-2.0",
+    "dependencies": {
+        "cordova-android": "^7.0.0",
+        "cordova-plugin-whitelist": "^1.3.3"
+    },
+    "cordova": {
+        "plugins": {
+            "cordova-plugin-whitelist": {}
+        },
+        "platforms": [
+            "android"
+        ]
+    }
+}
\ No newline at end of file
diff --git a/react-app/package.json b/react-app/package.json
index 3c10691..79270da 100644
--- a/react-app/package.json
+++ b/react-app/package.json
@@ -1,6 +1,7 @@
 {
   "name": "react-app",
   "version": "0.1.0",
+  "homepage": "./",
   "private": true,
   "dependencies": {
     "react": "^16.3.2",
@@ -13,4 +14,4 @@
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject"
   }
-}
\ No newline at end of file
+}
```

**Comment by leanbalma** [comment]
This works great now. I only had to run `cordova platform add android` before `npm run run android`, but that's OK. You can maybe add it to your script later if you want.

JS tip of the day
In ES6 you can write things like this:
```
execSync("cordova " + process.argv[2] + " " + process.argv[3]);
```
 as:
```
execSync(`cordova ${process.argv[2]} ${process.argv[3]}`);
```
:)

**Comment by fran-costa** [comment]
I added the changes suggested by @leanbalma in last comment to have a better boilerplate and to have to do a rebase =)

**Comment by leanbalma** [comment]
Closes #1.

**Pull request merged**